docs/blog/2020-01-23-why-typescript-chose-gatsby/index.md

```diff
@@ -0,0 +1,189 @@
+import { Layout } from "../components/layout"
+
+// Style
+import "./tsconfig.scss"
+
+// The main React component
+const TSConfigReferenceTemplateComponent = (props: PropTypes) => {
+  useEffect(() => {
+    // code which happens when the page has finished loading
+  })
+
+  // creation of static HTML via React + JSX
+  const post = props.data.markdownRemark
+  return (
+    <Layout locale={props.pageContext.locale}>
+      <div className="tsconfig">
+        <div id="full-option-list" className="indent">
+          <h1>TSConfig Reference</h1>
+        </div>
+
+        <nav id="sticky">
+          <ul>
+            <li>...</li>
+          </ul>
+        </nav>
+
+        <div className="indent">
+          <div dangerouslySetInnerHTML={{ __html: post.html! }} />
+        </div>
+      </div>
+    </Layout>
+  )
+}
+
+export default TSConfigReferenceTemplateComponent
+
+// The optional GraphQL query to get info for that page
+export const pageQuery = graphql`
+  query TSConfigReferenceTemplate($path: String, $tsconfigMDPath: String!) {
+    sitePage(path: { eq: $path }) {
+      id
+    }
+    markdownRemark(fileAbsolutePath: { eq: $tsconfigMDPath }) {
+      html
+    }
+  }
+`
+```
+
+I like this, because the rest of the team don't need to _learn_ React - just JSX which acts as a proxy for HTML.
+
+In the meanwhile, anyone working on the code stills get the advantages in tooling, because all of this is supported by the TypeScript compiler:
+
+- I can still make custom React components like `Layout` above to encapsulate complexity,and TypeScript still has the props verified
+- JSX support with TypeScript is dreamy, and I love it
+- The 'runtime' code inside `useEffect` will be transpiled and verified to be correct by TypeScript
+- The 'runtime' code can re-use the same libraries as the server-side rendering, but have strong guarantees about only being optionally available
+
+By not using any of the React `setState`-ish APIs, I can guarantee there is no "runtime" React rendering happening on a user's browser either. This means the HTML in the built file is exactly what someone will see whether they have JavaScript enabled or not. One advantage of this, has been that I can reliably run [`BackstopJS`](https://garris.github.io/BackstopJS/) to take screenshots of these static files to keep track of visual regressions as the site grows and others start to contribute.
+
+Would I recommend this constraint to people making Gatsby websites? Probably not - it's going against the grain (React is a really good tool) of how you're expected to use Gatsby. But the trade-off is worth it for me, and it lowers the barrier if a compiler engineer wants to contribute.
+
+### Speed
+
+I'm blown away by how fast Gatsby is for a user.
+
+The founder of Gatsby, Kyle Matthews [gave a great talk in 2017 on the ways in which Gatsby is fast](https://youtu.be/Gtd-Ht-D0sg?t=961) and [here more recently](https://www.youtube.com/watch?v=HQEotVfTXwk), in rough:
```

Comment:
```suggestion
The founder of Gatsby, Kyle Mathews [gave a great talk in 2017 on the ways in which Gatsby is fast](https://youtu.be/Gtd-Ht-D0sg?t=961) and [here more recently](https://www.youtube.com/watch?v=HQEotVfTXwk), in rough:
```

His last name is [Mathews](https://twitter.com/kylemathews) - I got caught on this one